### PR TITLE
Drop support for v0.10.  Use RequestInterrupt()

### PR DESCRIPTION
**Pull request by evantorrie**
RequestInterrupt can safely be called from another thread.  The
Interrupt callback cannot start the v8 interpreter, but it can use
such routines as v8::StackTrace::CurrentStackTrace.

This allows us to reinstate the stack backtrace feature without a performance penalty.

### CODE REVIEW

**Comment by yahoocla** [comment]
CLA is valid!

File: .travis.yml [hunk 1]
```diff
@@ -9,6 +9,6 @@ addons:
       - g++-4.9
 env: CXX=g++-4.9 CC=gcc-4.9
 node_js:
-    - "0.10"
     - "0.12"
-    - "4.0"
+    - "4"
+    - "6"
```
Comment (sylvio) [inline .travis.yml]: should we add "7"? The old code is not compiling with node7

File: .travis.yml [hunk 2]
```diff
@@ -9,6 +9,6 @@ addons:
       - g++-4.9
 env: CXX=g++-4.9 CC=gcc-4.9
 node_js:
-    - "0.10"
     - "0.12"
```
Comment (sylvio) [inline .travis.yml]: Node 0.12 has been EOLed. Maybe we can remove it from here too.

Comment (evantorrie) [inline .travis.yml]: I think I'll keep it here for a bit longer -- the API to make this work was added in v0.12, and it's "possible" there are still some people running it inside companies.

File: .travis.yml (hunk 1 above)
Comment (evantorrie) [inline .travis.yml]: OK, let me try with 7.

**Comment by evantorrie** [comment]
New code looks like it passes the unit tests on 7.x at least.

**Comment by sylvio** [comment]
LGTM

**Pull request merged**